src/HDF5.jl

```diff
@@ -989,7 +1017,14 @@ function generic_read(obj::DatasetOrAttribute, filetype::Datatype, ::Type{Opaque
 end
 
 # generic read function
+function generic_read!(buf::Union{AbstractMatrix{UInt8}, AbstractArray{T}}, obj::DatasetOrAttribute, filetype::Datatype, ::Type{T}, I...) where T
```

Comment:
Yes. Given that `generic_read` is defined in this package, we get to define the argument order. Since the buffer is the variable being altered, I put it first.